### DISCUSSION TITLE
Exclusion Filter not working?

### DISCUSSION THREAD

**Discussion by rubenporras** [spotbugs/spotbugs#1800] [Q&A] [answered]
Hi, 

I have an exclusion filter like

```xml
  <Match>
     <Bug pattern="MS_PKGPROTECT" />
     <Or>
       <Class name="~.*\.ModelInferenceSwitch" />
       <Class name="~.*\.ModelInferenceAdapterFactory" />
     </Or>
  </Match>
```

but Spotbugs still reports 

[INFO] com.avaloq.tools.ddk.xtext.modelinference.util.ModelInferenceAdapterFactory.modelPackage should be package protected [com.avaloq.tools.ddk.xtext.modelinference.util.ModelInferenceAdapterFactory] In ModelInferenceAdapterFactory.java MS_PKGPROTECT
[INFO] com.avaloq.tools.ddk.xtext.modelinference.util.ModelInferenceSwitch.modelPackage should be package protected [com.avaloq.tools.ddk.xtext.modelinference.util.ModelInferenceSwitch] In ModelInferenceSwitch.java MS_PKGPROTECT

Could you please assit me with the rules, please? I do not understand why they do not excude the finding for these clases.

I also have

```xml
  <Match>
     <Bug code="EI,EI2" />
     <Class name="~.*" />
  </Match>
```

but issues are still reported. Should that rule not just exclude this rules for any class?

Thanks a lot

**Comment by ThrawnCA** [comment, marked as answer]
If you want to exclude for every class, then you don't need the Class tag at all. Just exclude the bug pattern.

To exclude all members of a class, remember that the pattern is applied to the whole name, so you need to end with `.*`; see http://findbugs.sourceforge.net/manual/filter.html#d0e2197
